core/src/setup/Kokkos_Setup_Cuda.hpp

```diff
@@ -56,6 +56,10 @@
 #define KOKKOS_LAMBDA [=] __host__ __device__
 #define KOKKOS_CLASS_LAMBDA [ =, *this ] __host__ __device__
 
+#if defined(__clang__)
```

Comment:
```suggestion
#ifdef KOKKOS_COMPILER_CLANG
```

Reply:
Should we actually also add the same guard for the HIP backend ?

Reply:
It does not work on a CUDA clang compilation. My understanding is that the macro `KOKKOS_COMPILER_CLANG` is not yet defined at this stage.

Files in the core/src/setup directory should only depend on the backend, not the compiler right ?

Reply:
Oh you're right, it only gets defined later, these setup files get included at the top of the Kokkos_Macros.hpp header.  Sorry.